File: backend/internal/models/comment.go
package models

import "time"

// CommentFilter enthält die Paginierungsparameter für Kommentarlistenabfragen.
type CommentFilter struct {
	Page    int // Seitennummer (1-basiert)
	PerPage int // Einträge pro Seite
}

// CommentListItem repräsentiert einen einzelnen Kommentar in der öffentlichen Kommentarliste
// eines Anime.
type CommentListItem struct {
	ID         int64     `json:"id"`
	AnimeID    int64     `json:"anime_id"`
	AuthorName string    `json:"author_name"`
	Content    string    `json:"content"`
	CreatedAt  time.Time `json:"created_at"`
}

// CommentCreateInput enthält die Eingabedaten zum Erstellen eines neuen Kommentars.
type CommentCreateInput struct {
	AuthorName string
	Content    string
}

File: frontend/src/types/comment.ts
import { PaginationMeta } from '@/types/anime'

export interface CommentListItem {
  id: number
  anime_id: number
  author_name: string
  content: string
  created_at: string
}

export interface PaginatedCommentResponse {
  data: CommentListItem[]
  meta: PaginationMeta
}

export interface CommentCreateRequest {
  content: string
}

export interface CommentCreateResponse {
  data: CommentListItem
}
